### ISSUE TITLE
TestServer.Create<T> Startup does not get activated with correct services

### ISSUE THREAD

**Issue by HaoK** [aspnet/Hosting#221]
We regressed this as part of the latest hosting changes.

```
        public Startup(IHostingEnvironment env) { } => BOOM
```

cc @lodejard @davidfowl

**Comment by davidfowl** [comment]
Turns out this does work.

**Comment by HaoK** [comment]
Are you sure?  I wrote a simple test in hosting that also failed. I just added IHostingEnv to one of the startup classes and it failed. The template tests as checked in. are cheating and not using hosting engine so you cannot rely on them.

**Comment by davidfowl** [comment]
I'm writing that test right now using WebHost.Create and it's passing

**Comment by HaoK** [comment]
Okay, so its the `TestServer.Create<T>` with a startup type that's busted, since TestServer was new'ing up the service provider with its own StartupLoader to load the type, that's not following the normal flow and doesn't see AdditionalServices added to hosting engine.  Fixing templates to use a different overload for now

**Comment by HaoK** [comment]
Fixed as part of 234bbf82f2329e1ce2092962264ba5d4567d115b
